File: content/snippets/scala3/language_basics_conditions.scala
object ConditionsBasics:
  def gradeToLabel(grade: Int): String =
    if (grade >= 90) "excellent"
    else if (grade >= 75) "good"
    else if (grade >= 60) "pass"
    else "fail"

  @main def main(): Unit =
    val grade = 82
    val code = 2

    println(s"if/else: ${gradeToLabel(grade)}")

    code match {
      case 1 => println("match: created")
      case 2 => println("match: updated")
      case 3 => println("match: deleted")
      case _ => println("match: unknown")
    }

    val message = if (grade >= 60) "can continue" else "needs retry"
    println(s"if-expression: $message")

File: content/snippets/go/language_basics_conditions.go
package main

import "fmt"

func gradeToLabel(grade int) string {
	if grade >= 90 {
		return "excellent"
	} else if grade >= 75 {
		return "good"
	} else if grade >= 60 {
		return "pass"
	}
	return "fail"
}

func main() {
	grade := 82
	code := 2

	fmt.Println("if/else:", gradeToLabel(grade))

	switch code {
	case 1:
		fmt.Println("switch: created")
	case 2:
		fmt.Println("switch: updated")
	case 3:
		fmt.Println("switch: deleted")
	default:
		fmt.Println("switch: unknown")
	}

	ready := grade >= 60
	if ready {
		fmt.Println("decision: can continue")
	} else {
		fmt.Println("decision: needs retry")
	}
}
